leo/commands/build.rs

```diff
@@ -60,6 +62,18 @@ pub struct BuildOptions {
     pub enable_canonicalized_ast_snapshot: bool,
     #[structopt(long, help = "Writes AST snapshot after the type inference phase.")]
     pub enable_type_inferenced_ast_snapshot: bool,
+    #[structopt(long, help = "Writes all ASG debug graphs for the different compiler phases.")]
+    pub enable_all_asg_debug_graphs: bool,
+    #[structopt(long, help = "Writes ASG debug graph before the ASG compiler phases.")]
+    pub enable_initial_asg_debug_graph: bool,
+    #[structopt(long, help = "Writes ASG debug graph after the constants folding phase.")]
+    pub enable_constants_folded_asg_debug_graph: bool,
+    #[structopt(long, help = "Writes ASG debug graph after the dead code elimination phase.")]
+    pub enable_dead_code_eliminated_asg_debug_graph: bool,
+    #[structopt(long, help = "Edges to exclude from ASG debug graph.")]
+    pub asg_exclude_edges: Vec<String>,
+    #[structopt(long, help = "Node labels to exclude from ASG debug graph.")]
+    pub asg_exclude_labels: Vec<String>,
```

Comment:
the trait bound `Box<str>: FromStr` is not satisfied

Reply:
That's unfortunate. Sounds like a missing impl in the standard library that should exist. You can create a newtype wrapper `struct BoxStr(Box<str>);` for now though. Otherwise, see https://docs.rs/structopt/0.3.25/structopt/#custom-string-parsers. You should be able to use `#[structopt(short, parse(from_str))]` to use `impl From<String> for Box<str>`.

Reply:
Ah, I should've been a little more thorough. Thanks :)